openshift/console
# ODC-7234: Provide option to configure PAC on importing app via import from git flow

**Fixes**: 
https://issues.redhat.com/browse/ODC-7234

**Solution Description**: 
Added PAC Type in Pipelines Section of Git Import Flow. If the repository contains a `.tekton` folder, this type will be automatically selected. On creating, alongside the pre-existing resource creations, a Pipeline Repository will also be created. 

**ACs Completed:**

- [x] Remove Pipeline-as-Code from the import strategy section
- [x] If Pipelines operator is installed and PAC is enabled and .tekton dir is detected in the git repo then show Add Pipelines checkbox checked by default.
- [x] Below the checkbox show two radio buttons as Configure PAC (checked by default) and Add pipeline from this cluster(should be same as what we have today). 
- [x] If Pipelines operator is installed and PAC is not enabled or .tekton dir is not detected in the git repo then the UI should look like what we have today.
- [x] Show the application field in general section as of today
- [x] Show the resources section as today
- [x] Show advanced options as of today
- [x] On create, an imagestream, workload(depending on the resource that is selected), buildconfig, repository, and other resources (service/route) should be created like we do today
- [x] On create the user should navigate to the topology view as today

**Screen shots / Gifs for design review**: 

https://user-images.githubusercontent.com/47265560/213005074-9b471907-6518-42be-b38a-25d57afa6199.mp4

**Import Form when Builder Image Strategy is auto-selected when a repository with `.tekton` folder is used**

![screencapture-localhost-9000-import-ns-avik-2023-01-18-00_40_38](https://user-images.githubusercontent.com/47265560/213005354-3b87db8d-194c-4d4c-8858-7fb2370df070.png)

**Import Form when Docker Strategy is manually selected when a repository with `.tekton` folder is used**

![screencapture-localhost-9000-import-ns-avik-2023-01-18-00_41_49](https://user-images.githubusercontent.com/47265560/213005437-356b7206-58fe-4f95-a380-304e4d04b2d8.png)

**Import Form when a repository without `.tekton` folder is used**

![screencapture-localhost-9000-import-ns-avik-2023-01-18-01_57_46](https://user-images.githubusercontent.com/47265560/213005725-2815a74e-a75a-4fe0-abe2-b53a249e9ca3.png)

**Use Github App Tile Selected:**
![screencapture-localhost-9000-import-ns-avik-2023-01-18-02_19_24](https://user-images.githubusercontent.com/47265560/213009546-0cb752a3-6c05-4ed2-a7a7-6ebca8a84444.png)

**Setup Webhook Tile Selected:**
![screencapture-localhost-9000-import-ns-avik-2023-01-18-02_19_56](https://user-images.githubusercontent.com/47265560/213009602-6e41e23e-629a-4167-be0b-b0b5bc309415.png)


**Unit test coverage report**: 

E2E tests added.
Unit tests fixed.

**Test setup:**

1. Install the Pipelines Operator version above 1.8 (This comes with PAC pre-installed, which is a prerequisite)
2. Go to the Git Import Add Form.
3. Paste a Git URL having a .tekton folder.
a. GitHub: https://github.com/Lucifergene/oc-pipe
b. GitLab: https://gitlab.com/avikkundu/oc-pipe
c: Bitbucket: https://bitbucket.org/avikkundu/oc-pipe
4. You will see the Add Pipelines checkbox in the Pipelines Section is checked and Configure PAC option is selected.
a. If GitHub App is set up, Use GitHub app and Setup a webhook options, else only the Setup a webhook Form is shown.
5. Fill in the Git Secret and select a Webhook Secret. Optionally, edit the Name of the workload.
9. If you have already created a repository with the URL you have entered as the Git URL, you will get a backend error while submitting.

**Browser conformance**:
